servers/visual/rasterizer.h

```diff
@@ -37,6 +37,22 @@
 
 #include "core/self_list.h"
 
+#ifdef DEBUG_ENABLED
+#define SECTION_START(X) VSG::rasterizer->push_label(X)
+#define SECTION_END() VSG::rasterizer->pop_label()
+
+#define TIMESTAMP_FRAME_START() VSG::rasterizer->begin_frame_timings()
+#define TIMESTAMP(X) VSG::rasterizer->timestamp(X)
+#define TIMESTAMP_END() VSG::rasterizer->timestamp("END_TIMESTAMP") // necessary to calculate the time of the previous timestamp
+#else
+#define SECTION_START(X)
+#define SECTION_END()
+
+#define TIMESTAMP_FRAME_START()
+#define TIMESTAMP(X)
+#define TIMESTAMP_END()
+#endif
+
```

Comment:
These macros names could maybe be more informative? A reader seeing them would have to refer back to work out what they are doing.

Maybe e.g. `GPU_LOG_TIMESTAMP` or `PROFILER_SECTION_START` something like that.

Also less likely to have name collisions - it may not be a good idea to put these in `rasterizer.h` which is included everywhere, maybe a separate header only to be included from the necessary cpps may be better? And `#undef` the macros at the end would be cleanest, but you might get away with it providing the macro names are unique enough. Beware global namespace pollution! :grin: